### PR TITLE
better way handling response

### PR DESCRIPTION
**Pull request by honnix**
@danielnorberg PTAL

Fix the same mistake I made when handling response.

### CODE REVIEW

File: styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java [hunk 1]
```diff
@@ -102,19 +102,19 @@ public BackfillResource(String schedulerServiceBaseUrl, Storage storage) {
         Route.with(
             em.serializerResponse(BackfillPayload.class),
             "GET", BASE + "/<bid>",
-            rc -> getBackfill(arg("bid", rc))),
```
Comment (bergman) [inline styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java]: please inline instead, the argument is so far away from the method

Comment (danielnorberg) [inline styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java]: 👍

File: styx-api-service/src/main/java/com/spotify/styx/api/StyxConfigResource.java [hunk 2]
```diff
@@ -82,7 +82,7 @@ public StyxConfigResource(Storage storage) {
           Status.BAD_REQUEST.withReasonPhrase("Missing 'enabled' query parameter"));
     }
 
-    final boolean enabled = query.map(Boolean::parseBoolean).get();
+    final boolean enabled = Boolean.parseBoolean(query.get());
```
Comment (danielnorberg) [inline styx-api-service/src/main/java/com/spotify/styx/api/StyxConfigResource.java]: The `query` variable name is a bit confusing. First I thought it was the entire query string and wondered why the query string itself was being parsed as a boolean.

File: styx-api-service/src/main/java/com/spotify/styx/api/deprecated/WorkflowResource.java [hunk 3]
```diff
@@ -115,10 +113,10 @@ public WorkflowResource(com.spotify.styx.api.WorkflowResource workflowResource)
       Request request) {
     final Response<List<com.spotify.styx.model.data.WorkflowInstanceExecutionData>>
         response = workflowResource.instances(componentId, endpointId, request);
-    return response.payload()
-        .map(l -> l.stream().map(WorkflowInstanceExecutionData::create).collect(Collectors.toList()))
-        .map(Response::forPayload)
-        .orElse(Response.forStatus(response.status()));
+    return response.withPayload(response.payload()
+                                    .map(l -> l.stream().map(WorkflowInstanceExecutionData::create)
+                                        .collect(Collectors.toList()))
```
Comment (danielnorberg) [inline styx-api-service/src/main/java/com/spotify/styx/api/deprecated/WorkflowResource.java]: Indentation?

File: styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java (hunk 1 above)
Comment (honnix) [inline styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java]: will fix

File: styx-api-service/src/main/java/com/spotify/styx/api/StyxConfigResource.java (hunk 2 above)
Comment (honnix) [inline styx-api-service/src/main/java/com/spotify/styx/api/StyxConfigResource.java]: will fix

File: styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java (hunk 1 above)
Comment (honnix) [inline styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java]: inlined `arg()` if only one name

**Comment by bergman** [comment]
👍

**Pull request merged**